### PR TITLE
Add Notices Documentation to Storybook and upgrade to Storybook 7

### LINKED ISSUE #10681: Add notifications style guide to Storybook
We have _Shopper notifications usage guidelines_ (link in resource below) defined  and we need to publish them for the community.

We need to add them to the corresponding Storybook pages: [Notice Banner component](https://href.li/?https://woocommerce.github.io/woocommerce-blocks/?path=/docs/woocommerce-blocks-base-components-noticebanner--default) and [SnackbarList component](https://href.li/?https://woocommerce.github.io/woocommerce-blocks/?path=/docs/woocommerce-blocks-base-components-snackbarlist--default).

See pdFofs-11n-p2 P2 for more details.

### PR DESCRIPTION
**Pull request by mikejolley**
## What

This PR upgrades Storybook to v7 and adds usage guidelines in the form of MDX files for NoticeBanner and SnackbarList components. It was necessary to bump versions of several packages to support Storybook 7, and modify existing stories for compatibility. 

As well as the upgrade and new MDX files, I reorganised existing stories so that the sidebar is easier to navigate. I've gone through each story to ensure there are no errors.

Fixes #10681

## Why

Storybook 7 has improved UI and supports inline documentation (MDX files) for documenting our components with markdown.

## Testing Instructions

1. Install npm dependencies and successfully build blocks. There should be no errors with the build.
2. Run `npm run storybook` and check for successful build.
3. Navigate to storybook. Go through each item in the navigation and check for errors.
4. Go to Base Components > NoticeBanner > Docs. proof.
5. Go to Base Components > SnackbarList > Docs. proof.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

Screenshot 2023-10-31 at 15 44 04
Screenshot 2023-10-31 at 15 43 53

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [ ] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

N/A

### CODE REVIEW

File: assets/js/base/components/notice-banner/docs/docs.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+### Default Notices
+
+By default, noices are grey and used for less important messaging.
+
+<Canvas of={ NoticeBannerStories.Default } />
+
+### Informational Notices
+
+Blue notices with an info icon are used for general information for buyers that are not blocking and do not require action.
```
Comment (alexflorisca) [inline assets/js/base/components/notice-banner/docs/docs.mdx]:
```suggestion
Blue notices with an info icon are used for general information for buyers, but do not require them to take an action.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 2]
```diff
@@ -1,62 +1,89 @@
...
-A temporary informational UI displayed at the bottom of store pages.
+import * as SnackbarListStories from '../stories/index.stories.tsx';
+import * as SnackbarStories from '../stories/snackbar.stories.tsx';
+
+<Meta name="Docs" of={ SnackbarListStories } />
+
+# SnackbarList
 
-## Table of contents <!-- omit in toc -->
+A temporary informational UI displayed at the bottom of store pages.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
A temporary informational UI element displayed at the bottom of store pages.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 3]
```diff
@@ -1,62 +1,89 @@
...
-        -   [`notices`: `NoticeType[]`](#notices-noticetype)
+<Canvas
+	of={ SnackbarListStories.Default }
+	layout="padded"
+	className={ 'force-canvas-height' }
+/>
 
 ## Design Guidelines
 
 The buyer notice snackbar is temporary informational UI displayed at the bottom of store pages. WooCommerce blocks, themes, and plugins all use snackbar notices to indicate the result of a successful action.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
The Snackbar is a temporary informational UI element displayed at the bottom of store pages. WooCommerce blocks, themes, and plugins all use snackbar notices to indicate the result of a successful action. For example, adding something to the cart.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 4]
```diff
@@ -1,62 +1,89 @@
...
 
 Snackbar notices work in the same way as the NoticeBanner component, and support the same statuses and styles.
 
-## Development Guidelines
+### Default Snackbars
 
-### Usage
+By default, noices are grey and used for less important messaging.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
By default, notices are grey and used for less important messaging.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 5]
```diff
@@ -1,62 +1,89 @@
...
 
-const notices = [
-	{
-        id: '1',
-        content: 'This is a snackbar notice.',
-        status: 'default',
-        isDismissible: true,
-    }
-];
+Blue notices with an info icon are used for general information for buyers that are not blocking and do not require action.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
Blue notices with an info icon are used for general information for buyers, but do not require action.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 6]
```diff
@@ -1,62 +1,89 @@
...
+
+<ArgTypes />
 
-#### `notices`: `NoticeType[]`
+### NoticeType
 
-A list of notices to display as snackbars. Each notice must have an `id` and `content` prop.
+Each notice can be provided with the following args.
 
 -   The `id` prop is used to identify the notice and should be unique.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
-   The `id` (required) prop is used to identify the notice and should be unique.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 7]
```diff
@@ -1,62 +1,89 @@
...
 -   The `content` prop is the content to display in the notice.
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx]:
```suggestion
-   The `content` (required) prop is the content to display in the notice.
```

File: assets/js/base/components/snackbar-list/docs/docs.mdx [hunk 8]
```diff
@@ -1,62 +1,89 @@
...
+
+```jsx
+import { SnackbarList } from '@woocommerce/base-components';
+
+const notices = [
+	{
+        id: '1',
+        content: 'This is a snackbar notice.',
+        status: 'default',
+        isDismissible: true,
```
Comment (alexflorisca) [inline assets/js/base/components/snackbar-list/docs/docs.mdx:84]:
```suggestion
        isDismissible: true,
        spokenMessage: "Hello snackbar!"
```

Comment (alexflorisca) [reply assets/js/base/components/snackbar-list/docs/docs.mdx:84]: Might as well include all the available props in the example.

File: storybook/main.js [hunk 9]
```diff
@@ -1,26 +1,38 @@
...
 		'@storybook/addon-links',
 		'storybook-addon-react-docgen',
-		'@storybook/addon-postcss',
+		'@storybook/addon-styling-webpack',
 	],
+	typescript: {
+		reactDocgen: 'react-docgen-typescript-plugin',
+	},
+	docs: {
+		//👇 See the table below for the list of supported options
```
Comment (alexflorisca) [inline storybook/main.js]:
```suggestion
```

**Review by alexflorisca — APPROVED** [review]
Awesome stuff @mikejolley, this is a ABIG win, thanks for working on it! I've left some minor suggestions to do with text corrections and an uneccessary comment, but I will approve as they are non blocking. Great stuff.

**Review by opr — APPROVED** [review]
Works for me, thanks for taking care of the failing tests too!

**Comment by ralucaStan** [comment]
Amazing work on this one!

**Pull request merged**